src/zope/i18n/tests/test_compile.py

```diff
@@ -0,0 +1,68 @@
+##############################################################################
+#
+# Copyright (c) 2017 Zope Foundation and Contributors.
+# All Rights Reserved.
+#
+# This software is subject to the provisions of the Zope Public License,
+# Version 2.1 (ZPL).  A copy of the ZPL should accompany this distribution.
+# THIS SOFTWARE IS PROVIDED "AS IS" AND ANY AND ALL EXPRESS OR IMPLIED
+# WARRANTIES ARE DISCLAIMED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF TITLE, MERCHANTABILITY, AGAINST INFRINGEMENT, AND FITNESS
+# FOR A PARTICULAR PURPOSE.
+#
+##############################################################################
+
+import unittest
+
+from zope.i18n import compile
+
+
+@unittest.skipUnless(compile.HAS_PYTHON_GETTEXT,
+                     "Need python-gettext")
+class TestCompile(unittest.TestCase):
+
+    def test_non_existant_path(self):
+
+        self.assertIsNone(compile.compile_mo_file('no_such_domain', ''))
+
+    def test_po_exists_but_invalid(self):
+        import tempfile
+        import shutil
+        import os.path
+
+        td = tempfile.mkdtemp()
+        self.addCleanup(shutil.rmtree, td)
+
+        with open(os.path.join(td, "foo.po"), 'w') as f:
+            f.write("this should not compile")
+
+        self.assertEqual(
+            0,
+            compile.compile_mo_file('foo', td))
```

Comment:
Besides, if you were using 0 vs False to distinguish outcomes in the unit tests, well, `0 == False` so ...